roda-core/roda-core/src/main/java/org/roda/core/plugins/base/synchronization/instanceIdentifier/InstanceIdentifierPreservationAgentPlugin.java

```diff
@@ -136,14 +140,32 @@ private void modifyInstanceId(ModelService model, IndexService index, Job cached
 
     Report reportItem = PluginHelper.initPluginReportItem(this, cachedJob.getId(), Job.class);
     PluginHelper.updatePartialJobReport(this, model, reportItem, false, cachedJob);
+    DistributedInstances distributedInstances = null;
+    List<String> distributedInstanceIds = new ArrayList<>();
+
+    try {
+      distributedInstances = RodaCoreFactory.getModelService().listDistributedInstances();
+    } catch (RequestNotValidException | GenericException | AuthorizationDeniedException | IOException e) {
+      LOGGER.error("Could not list distributed instances");
+    }
+
+    if (distributedInstances != null) {
+      distributedInstanceIds = distributedInstances.getObjects().stream()
+        .map(distributedInstance -> distributedInstance.getId()).collect(Collectors.toList());
+    }
 
     try (CloseableIterable<OptionalWithCause<PreservationMetadata>> iterable = model.listPreservationAgents()) {
       for (OptionalWithCause<PreservationMetadata> opm : iterable) {
         try {
           if (opm.isPresent()) {
             PreservationMetadata pm = opm.get();
-            PremisV3Utils.updatePremisUserAgentId(pm, model, index, instanceId);
-            countSuccess++;
+            if (URNUtils.hasInstanceId(pm.getId())) {
+              String instanceId = URNUtils.extractInstanceIdentifierFromId(pm.getId());
+              if (!distributedInstanceIds.contains(instanceId)) {
+                PremisV3Utils.updatePremisUserAgentId(pm, model, index, RODAInstanceUtils.getLocalInstanceIdentifier());
+                countSuccess++;
+              }
+            }
```

Comment:
You are not considering the logic in the local instance. In the local instance there will always be the error on line 149 and on line 162 you are preventing the addition of instanceID for items without instanceID.